src/main/java/com/iexec/common/sdk/marketplace/MarketplaceClient.java

```diff
@@ -78,6 +78,32 @@ public List<WorkerpoolMarketOrder> getWorkerpoolOrders(String workerpoolAddress)
         return getOrders(OrderType.WORKERPOOL, queryParams);
     }
 
+    @Override
+    public List<WorkerpoolMarketOrder> getTaggedWorkerpoolOrders(String tag) {
+        final MPQueryParams queryParams = MPQueryParams.builder()
+                .minTag(tag)
+                .maxTag(tag)
+                .build();
+        return getOrders(OrderType.WORKERPOOL, queryParams);
+    }
+
+    @Override
+    public Optional<WorkerpoolMarketOrder> getOneTaggedWorkerpoolOrder(String workerpoolAddress, String tag) {
```

Comment:
It is really minor, but shouldn't it be better to only query the marketplace for a single order instead of fetching several orders to only keep one ?

Reply:
I don't think the marketplace exposes an endpoint to fetch a single order.

Reply:
No, it isn't

Reply:
Could you explain why ?

Reply:
https://github.com/iExecBlockchainComputing/iexec-market-api/blob/c2ec2d1ad44b50a487ce77bac0d41afbbc847502/api/src/controllers/router.js#L420

Reply:
I get it, your answers were not about knowing if it's better or not, it is just not possible with the current API which returns a paged result with fixed parameters.
In Spring, there is a really elegant implementation which allows to dynamically size a request with a `PageRequest`. It is possible to configure paging and sorting parameters before a JPA request execution.

Reply:
Yes, I checked that kind of behavior but page here is just an int right now, not a object :( !